Source/RapyutaSimulationPlugins/Private/Core/RRNetworkPlayerController.cpp

```diff
@@ -183,10 +188,15 @@ APawn* ARRNetworkPlayerController::FindPawnToPossess()
     for (auto& entity : ServerSimState->EntityList)
     {
         matchingPawn = Cast<APawn>(entity);
-        if (IsValid(matchingPawn) && (nullptr == matchingPawn->GetController()) && (GetPawn() != matchingPawn))
+        ARRBaseRobot* robot = Cast<ARRBaseRobot>(entity);
+        if (IsValid(robot) && !robot->isPossessed && (GetPawn() != robot))
         {
             break;
         }
+        else
+        {
+            matchingPawn = nullptr;
+        }
```

Comment:
`matchingPawn = Cast<APawn>(entity)` could be moved to before `break`, so we would not need to reassign it to `nullptr`?